Partial Display Refresh Support

skmcgrail opened the pull request:
Adds partial display refresh support based on the Pimoroni C implementation.

skmcgrail committed:
Partial Display Refresh Support

9names commented:
Hi, thanks for the PR!
Have you tested this? It doesn't seem to be working for me, I tried with
```rust
let repaint_region = Rectangle::new(Point::new(32, 32), Size::new(32, 32));
let _ = display.partial_update(repaint_region.into());
```

According to the docs for the micropython driver the only restriction is that the Y values must be a multiple of 8, which the above should be doing fine.
Also, we should probably document this requirement, and maybe enforce it.

skmcgrail committed:
Fix transfer range, change to TryFrom trait implementation, add boundary checks to DisplayRegion

skmcgrail commented:
Updated the pull request to implement the feedback on checking the boundaries on the Y/Height values, and added relevant documentation.

I also uploaded a test driver that you can use test it. This will iterate all 16 channels of the display (128/8) to show off the partial refresh for each vertical segment. You can find that [here](https://github.com/skmcgrail/badger2040-partial-refresh). I've only tested this with a badger2040 (only device I have with this screen), so if you have another device that has smaller or full-size resolution would be good to verify as well.

9names reviewed (approved):
The only compatible screen I have is a badger2040 as well.
This works with my test program, and your example is a good showcase of this feature too.
LGTM
